test/parallel_api/iterator/permutation_iterator.pass.cpp

```diff
@@ -39,13 +81,490 @@ main()
+                generate_data(host_keys.get(), host_keys.get() + n);
+        
+                const auto result = dpl::transform_reduce(exec, permItBegin, permItEnd, TestValueType{},
+                                                            ::std::plus<TestValueType>(), ::std::negate<TestValueType>());
+                // TODO wait_and_throw(exec);
+
+                const auto expected =
+                    ::std::transform_reduce(permItBegin, permItEnd, TestValueType{}, ::std::plus<TestValueType>(),
+                                            ::std::negate<TestValueType>());
+       
+                EXPECT_EQ(expected, result, "Wrong result of dpl::transform_reduce");
+            });
+    }
+
+    template <typename Policy, typename Iterator1, typename Size>
+    ::std::enable_if_t<!is_random_access_iterator_v<Iterator1>>
+    operator()(Policy&& exec, Iterator1 first1, Iterator1 last1, Size n)
+    {
+    }
+};
+
+// dpl::merge, dpl::inplace_merge -> __parallel_merge
+DEFINE_TEST(test_merge)
+{
+    DEFINE_TEST_CONSTRUCTOR(test_merge)
+
+    template <typename TIterator>
+    void generate_data(TIterator itBegin, TIterator itEnd, TestValueType initVal)
+    {
+        const auto testing_n = ::std::distance(itBegin, itEnd);
+    
+        TestValueType val = initVal;
+        for (auto it = itBegin; it != itEnd; ++it)
+            *it = val++;
+    }
+
+    template <typename Policy, typename Iterator1, typename Iterator2, typename Size>
+    ::std::enable_if_t<is_random_access_iterator_v<Iterator1>>
+    operator()(Policy&& exec, Iterator1 first1, Iterator1 last1, Iterator2 first2, Iterator1 last2, Size n)
+    {
+        TestDataTransfer<UDTKind::eKeys, Size> host_keys(*this, n);     // source data(1) for merge
+        TestDataTransfer<UDTKind::eVals, Size> host_vals(*this, n);     // source data(2) for merge
+        
+        // Fill full source data set
+        generate_data(host_keys.get(), host_keys.get() + n, TestValueType{});
+        generate_data(host_vals.get(), host_vals.get() + n, TestValueType{} + n / 2);
+
+        test_through_permutation_iterator_mibt{first1, n}(
+            [&](auto permItBegin1, auto permItEnd1)
+            {
+                test_through_permutation_iterator_mibt{first2, n}(
+                    [&](auto permItBegin2, auto permItEnd2)
+                    {
+                        std::vector<TestValueType> resultBuf;
+                        resultBuf.resize(2 * n);
+
+                        const auto resultEnd = dpl::merge(exec, permItBegin1, permItEnd1, permItBegin2, permItEnd2, resultBuf.begin());
+                        // TODO wait_and_throw(exec);
+
+                        std::vector<TestValueType> expected;
+                        expected.resize(3 * n);
+
+                        auto expectedEnd = std::merge(permItBegin1, permItEnd1, permItBegin2, permItEnd2, expected.begin());
+
+                        const auto expectedSize = ::std::distance(expected.begin(), expectedEnd);
```

Comment:
Due yo we have `::std::enable_if_t<is_random_access_iterator_v<Iterator1>>`
I think we can write just `-` instead of `std::distance`.

Reply:
As described at https://en.cppreference.com/w/cpp/iterator/distance. there are specialization for this case.
An I prefer `std::distance`